web/packages/shared/components/TerminalSearch/TerminalSearch.tsx

```diff
@@ -0,0 +1,204 @@
+/**
+ * Teleport
+ * Copyright (C) 2024 Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { useEffect, useState, useRef, useCallback } from 'react';
+import styled, { useTheme } from 'styled-components';
+import { Flex, ButtonIcon, Box, P2, Input } from 'design';
+import { ChevronDown, ChevronUp, Cross } from 'design/Icon';
+import { SearchAddon } from '@xterm/addon-search';
+
+export interface TerminalSearcher {
+  getSearchAddon(): SearchAddon;
+  focus(): void;
+  registerCustomKeyEventHandler(customEvent: (e: KeyboardEvent) => boolean): {
+    unregister(): void;
+  };
+}
+
+export const TerminalSearch = ({
+  xTerm,
+  show,
+  onClose,
+  onOpen,
+  isSearchKeyboardEvent,
+}: {
+  xTerm: TerminalSearcher;
+  show: boolean;
+  onClose(): void;
+  onOpen(): void;
+  isSearchKeyboardEvent: (e: KeyboardEvent) => boolean;
```

Comment:
Nit: other functions in this type don't use the arrow notation, we can unify it.